addons/binding/org.openhab.binding.xiaomivacuum/src/main/java/org/openhab/binding/xiaomivacuum/handler/XiaomiVacuumHandler.java

```diff
@@ -0,0 +1,402 @@
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.google.gson.JsonArray;
+import com.google.gson.JsonObject;
+import com.google.gson.JsonParser;
+import com.google.gson.JsonSyntaxException;
+
+/**
+ * The {@link XiaomiVacuumHandler} is responsible for handling commands, which are
+ * sent to one of the channels.
+ *
+ * @author Marcel Verpaalen - Initial contribution
+ */
+public class XiaomiVacuumHandler extends BaseThingHandler {
+    private final Logger logger = LoggerFactory.getLogger(XiaomiVacuumHandler.class);
+
+    private ScheduledFuture<?> pollingJob;
+
+    private JsonParser parser;
+    private byte[] token;
+
+    private final long CACHE_EXPIRY = TimeUnit.SECONDS.toMillis(5);
+    private ExpiringCache<String> status;
+    private ExpiringCache<String> consumables;
+    private ExpiringCache<String> dnd;
+    private ExpiringCache<String> history;
+
+    private RoboCommunication roboCom;
+    private int lastId;
+
+    public XiaomiVacuumHandler(Thing thing) {
+        super(thing);
+        parser = new JsonParser();
+    }
+
+    @Override
+    public void handleCommand(ChannelUID channelUID, Command command) {
+        if (roboCom == null) {
+            logger.debug("Vacuum {} not online. Command {} ignored", getThing().getUID(), command.toString());
+            return;
+        }
+        if (command == RefreshType.REFRESH) {
+            logger.debug("Refreshing {}", channelUID);
+            updateData();
+            return;
+        }
+        if (channelUID.getId().equals(CHANNEL_VACUUM)) {
+            if (command instanceof OnOffType && command == OnOffType.ON) {
+                sendCommand(VacuumCommand.START_VACUUM);
+            } else if (channelUID.getId().equals(CHANNEL_VACUUM)) {
+                if (command instanceof OnOffType && command == OnOffType.OFF) {
+                    sendCommand(VacuumCommand.STOP_VACUUM);
+                }
+            }
+        }
+        if (channelUID.getId().equals(CHANNEL_SPOT)) {
+            if (command instanceof OnOffType && command == OnOffType.ON) {
+                sendCommand(VacuumCommand.START_SPOT);
+            } else if (channelUID.getId().equals(CHANNEL_VACUUM)) {
+                if (command instanceof OnOffType && command == OnOffType.OFF) {
+                    sendCommand(VacuumCommand.STOP_VACUUM);
+                }
+            }
+        }
+        if (channelUID.getId().equals(CHANNEL_PAUSE) && command instanceof OnOffType && command == OnOffType.ON) {
+            sendCommand(VacuumCommand.PAUSE);
+        }
+        if (channelUID.getId().equals(CHANNEL_RETURN) && command instanceof OnOffType && command == OnOffType.ON) {
+            sendCommand(VacuumCommand.STOP_VACUUM);
+            sendCommand(VacuumCommand.CHARGE);
+        }
+        if (channelUID.getId().equals(CHANNEL_COMMAND)) {
+            updateState(CHANNEL_COMMAND, new StringType(sendCommand(command.toString())));
```

Comment:
okay, got it (from your other commend wrt to feeding back the result from the command)